lit/docs/getting-started/resources.lit

```diff
@@ -0,0 +1,542 @@
+}
+
+\section{
+  \title{Put Steps}
+
+  The last piece of the resource interface is \reference{put-step}{put steps}.
+  Put steps are generally for pushing some change to the external system or
+  object the resource represents. What this means will vary by resource, so
+  again, you should carefully read the documentation for the specific resource
+  you are using. Each resource is basically its own little application.
+
+  Let's add a \reference{put-step} to our \code{hello-world.yml} pipeline.
+  We'll add another resource called \code{timestamps} that the
+  \reference{put-step} will emit to.
+
+  \codeblock{yaml}{{
+  resources:
+  - name: every-1min
+    type: time
+    source:
+      interval: 1m
+  # add a new resource
+  - name: timestamps
+    type: mock  # from https://github.com/concourse/mock-resource
+
+  jobs:
+  - name: hello-world-job
+    plan:
+    - get: every-1min
+      trigger: true
+    - task: hello-world-task
+      config:
+        platform: linux
+        image_resource:
+          type: registry-image
+          source:
+            repository: busybox
+        inputs:
+        - name: every-1min
+        run:
+          path: cat
+          args: ["every-1min/timestamp"]
+    # add a put step to the end of the job
+    - put: timestamps
+      params:
+        file: every-1min/timestamp
+  }}
+
+  Set the pipeline and let's watch it trigger and run from the \link{web
+  UI}{http://localhost:8080/teams/main/pipelines/hello-world}
+
+  \codeblock{}{{
+  $ fly -t tutorial set-pipeline -p hello-world -c hello-world.yml
+  }}
+
+  The job now has the \code{every-1min} resource to the left and the new
+  \code{timestamps} resource on the right. This is how Concourse visually
+  represents the inputs (\reference{get-step}{get steps}) and outputs
+  (\reference{put-step}{put steps}) of a job.
+
+  That covers the basics of resources. It is best of think of each resource as
+  a little application, so carefully read any documentation the resource author
+  has provided. Every resource behaves a little differently based on the
+  external system it is representing.
+}
+
+\section{
+  \title{Using External Resource Types}
+  Concourse comes bundled with a lot of resources that are enough for most
+  people to start using Concourse with. However, users will want to extend
+  Concourse to work with all sorts of systems and that means bringing your own
+  \reference{resource-types}.
+
+  Adding a \reference{resource-types}{resource type} to your pipeline looks
+  very similar to adding a resource. You can even override the bundled resource
+  types by re-declaring them in your pipeline.
+
+  Remember, a resource is a container image. So to pull in a new
+  resource type you need to tell Concourse \italic{where} to pull the image
+  from. This is done by using the built-in \link{registry-image
+  resource}{https://github.com/concourse/registry-image-resource/}.
+
+  Here is how we would re-declare the \code{time} resource in our
```

Comment:
This maybe isn't the most compelling example of why you'd want to use a custom resource type. Maybe we could use one from the community? There's a good example of using the RSS resource in https://concourse-ci.org/resource-types.html, which I think is a pretty intuitive example.